File: vacancy-app/src/main/java/a/gleb/vacancy_app/db/entity/VacancyEntity.java
package a.gleb.vacancy_app.db.entity;

import a.gleb.vacancy_app.model.enums.EmploymentType;
import a.gleb.vacancy_app.model.enums.ExperienceLevel;
import a.gleb.vacancy_app.model.enums.VacancyStatus;
import a.gleb.vacancy_app.model.enums.WorkFormat;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

@Getter
@SuperBuilder
@NoArgsConstructor
@Entity
@Table(name = "vacancy", schema = "vacancy")
public class VacancyEntity extends BaseEntity {

    @Column(name = "company_id", nullable = false)
    private UUID companyId;

    @Column(name = "title", nullable = false, length = 200)
    private String title;

    @Column(name = "description", columnDefinition = "TEXT")
    private String description;

    @Column(name = "requirements", columnDefinition = "TEXT")
    private String requirements;

    @Column(name = "salary_from")
    private Long salaryFrom;

    @Column(name = "salary_to")
    private Long salaryTo;

    @Column(name = "currency", length = 10)
    private String currency;

    @Enumerated(EnumType.STRING)
    @Column(name = "employment_type", length = 30)
    private EmploymentType employmentType;

    @Enumerated(EnumType.STRING)
    @Column(name = "work_format", length = 20)
    private WorkFormat workFormat;

    @Enumerated(EnumType.STRING)
    @Column(name = "experience_level", length = 20)
    private ExperienceLevel experienceLevel;

    @Column(name = "city", length = 100)
    private String city;

    @Column(name = "country", length = 100)
    private String country;

    @Enumerated(EnumType.STRING)
    @Column(name = "status", nullable = false, length = 20)
    private VacancyStatus status;

    @Column(name = "contact_name", length = 200)
    private String contactName;

    @Column(name = "contact_email", length = 200)
    private String contactEmail;

    @Column(name = "contact_phone", length = 50)
    private String contactPhone;

    @OneToMany(mappedBy = "vacancy", cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.LAZY)
    private List<VacancySkillEntity> skills = new ArrayList<>();

    @PrePersist
    @Override
    public void onCreate() {
        super.onCreate();
    }

    @PreUpdate
    @Override
    public void onUpdate() {
        super.onUpdate();
    }
}

File: main-service/src/main/kotlin/a/gleb/main_service/db/entity/vacancy/VacancyEntity.kt
package a.gleb.main_service.db.entity.vacancy

import a.gleb.main_service.db.entity.BaseEntity
import org.springframework.data.relational.core.mapping.Column
import org.springframework.data.relational.core.mapping.Table
import java.time.LocalDate

@Table(name = "vacancies")
class VacancyEntity(
    @Column(value= "title")
    var title: String,

    @Column(value= "description")
    var description: String,

    @Column(value= "company_name")
    var companyName: String,

    @Column(value= "contact_email")
    var contactEmail: String? = null,

    @Column(value= "salary_from")
    var salaryFrom: Int? = null,

    @Column(value= "salary_to")
    var salaryTo: Int? = null,

    @Column(value= "salary_currency")
    var salaryCurrency: String? = null,

    var requirements: MutableSet<String> = mutableSetOf(),

    var benefits: MutableSet<String> = mutableSetOf(),

    @Column(value= "employment_type")
    var employmentType: String? = null,

    @Column(value= "work_schedule")
    var workSchedule: String? = null,

    @Column(value= "experience_level")
    var experienceLevel: String? = null,

    @Column(value= "location")
    var location: String? = null,

    @Column(value= "is_remote")
    var isRemote: Boolean = false,

    @Column(value= "is_active")
    var isActive: Boolean = true,

    @Column(value= "application_deadline")
    var applicationDeadline: LocalDate? = null

) : BaseEntity()